### PR TITLE
backend: (riscv) use register_queue for register management

### PR DESCRIPTION
**Pull request by superlopuh**
Decouple the traversal logic from available register logic. This change just uses the helper class instead of having the logic inline, future changes will do a bit more separation.

### CODE REVIEW

File: xdsl/backend/riscv/register_allocation.py [hunk 1]
```diff
@@ -15,7 +16,7 @@ class RegisterAllocator(abc.ABC):
     Base class for register allocation strategies.
     """
 
-    def __init__(self, limit_registers: int | None = None) -> None:
+    def __init__(self) -> None:
         pass
```
Comment (AntonLydike) [inline xdsl/backend/riscv/register_allocation.py:20]: Empty init can be dropped?

**Review by AntonLydike — APPROVED** [review]
Minor critic, otherwise this PR :rock:s as well!

File: xdsl/backend/riscv/register_allocation.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/backend/riscv/register_allocation.py:20]: Probably, I'll take a look

**Review by nazavode — APPROVED** [review]

File: xdsl/backend/riscv/register_allocation.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/backend/riscv/register_allocation.py:20]: I'm not sure exactly what the rules are for abstract base classes and inits. What I'd like to communicate here is that all subclasses should implement an init with no parameters, but I don't think this is actually enforced. I'll delete it for now until I have a better idea.

**Pull request merged**